### PR TITLE
WIP: New 'profile' and 'announcement' commands

### PR DESCRIPTION
**Pull request by SandroHc**

### CODE REVIEW

File: app/modules/announcements/announcements.py [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+    def replace_message_template(self, template: str, default: str, member: DiscordMember, db_member: DatabaseMember):
+        if not template:
+            template = default
+
+        message = template.replace("{number}", str(db_member.lab_member_number))  # TODO: add zero-padding, e.g. 001 
+        message = message.replace("{name}", member.mention)
+        return message
+    
+    def create_member(self, member: DiscordMember):
+        db_member = get_member(gid=member.guild.id, uid=member.id)
```
Comment (MatBourgon) [inline app/modules/announcements/announcements.py]: You don't need to pass in named arguments for get_member and get_guild, but I'm seeing this everywhere. Is this some Pythonic thing I'm missing? I can understand for lines like:

  `embed.add_field(name="Lab Member", value=f"#{db_member.lab_member_number}", inline=False)`

As it helps with readability and parameter naming, but `get_member(gid=message.guild.id, uid=message.author.id)` is explicit even without the keywords

File: app/database/database.py [hunk 2]
```diff
@@ -19,10 +20,28 @@ def init():
         logger.critical(e)
         exit(1)
 
-def get_member(gid, uid):
+
+def get_guild(gid):
```
Comment (MatBourgon) [inline app/database/database.py]: To keep things consistent, get_guild should probably also have a create_if_doesnt_exist=True parameter

File: app/database/models/member.py [hunk 3]
```diff
@@ -2,7 +2,14 @@
...
-    uid   = LongField(required=True, primary_key=True, unique_with=["gid"])
-    xp    = LongField(required=True)
-    level = LongField(required=True)
+    gid           = LongField(null=False, required=True)
+    uid           = LongField(null=False, required=True, primary_key=True, unique_with=["gid"])
+    xp            = LongField(default=0)
+    level         = LongField(default=0)
+    display_name  = StringField()
+    name          = StringField()
+    discriminator = LongField()
```
Comment (MatBourgon) [inline app/database/models/member.py]: Do we really need to store the user's display name, username, and discriminator? That information can get obsolete very fast, and the effort to get the information from Discord isn't that hard (you typically will have a DiscordMember object already)

File: app/database/models/member.py [hunk 4]
```diff
@@ -2,7 +2,14 @@
...
+    joined_at     = DateTimeField()
+    lab_member_number = LongField()
```
Comment (MatBourgon) [inline app/database/models/member.py]: `lab_member_number` should probably be a `IntField` instead of `LongField`. I'm sure ~2-4 Million entries is enough (I'd propose `ShortField` but it doesn't look like it's a default type).

In the same vein (this isn't a you thing), but we can probably do the same for the `xp` and `level`. Since `xp` is the current xp between levels and not the total raw xp, we should be fine for a while (level 100 only requires 10 000 xp to rank up), and I doubt someone will hit level 2-4 million.

File: app/modules/core/error_handler.py [hunk 5]
```diff
@@ -34,7 +34,7 @@ async def on_command_error(self, ctx: Context, error: CommandError):
         """Commands error handler."""
         
         if ErrorHandler.is_unknown_error(error):
-            logger.error(error)
+            logger.exception(error)
```
Comment (MatBourgon) [inline app/modules/core/error_handler.py]: Why the change from error to exception?

**Review by MatBourgon — CHANGES_REQUESTED** [review]
See comments above:

File: app/modules/announcements/announcements.py (hunk 1 above)
Comment (SandroHc) [reply app/modules/announcements/announcements.py]: I don't generally remember the order of the arguments so I just let the IDE suggest them. I'll do a pass to clean it up

File: app/database/models/member.py (hunk 3 above)
Comment (SandroHc) [reply app/database/models/member.py]: The name was to visually identify a record, but I agree with you that it is currently bloat. I'll remove those

File: app/database/models/member.py (hunk 4 above)
Comment (SandroHc) [reply app/database/models/member.py]: Changed for `lab_member_number`. Do you want me to change for `xp` and `level` too?

File: app/modules/core/error_handler.py (hunk 5 above)
Comment (SandroHc) [reply app/modules/core/error_handler.py]: `exception()` prints the stacktrace too.

File: app/database/database.py (hunk 2 above)
Comment (SandroHc) [reply app/database/database.py]: Refactored to `Guild.get_guild()`, like what Shaade did on the Member class

File: app/database/models/member.py (hunk 4 above)
Comment (MatBourgon) [reply app/database/models/member.py]: If you haven't pushed it yet, that would be nice, otherwise it can be changed in a later commit.

Comment (SandroHc) [reply app/database/models/member.py]: Consider it done 🥼

**Review by MatBourgon — APPROVED** [review]
